Alignment/OfflineValidation/plugins/SagittaBiasNtuplizer.cc

```diff
@@ -0,0 +1,550 @@
+  double openingAngle = this->openingAngle(myTracks[0], myTracks[1]);
+  h_CosOpeningAngle->Fill(openingAngle);
+  passOpeningAngle = true;  //(openingAngle > M_PI/4.);
+
+  if (!passOpeningAngle)
+    return;
+  h_cutFlow->Fill(8);
+
+  if (doGen_) {
+    const edm::View<reco::Candidate>* genPartCollection = &event.get(genParticlesToken_);
+
+    // loop on the reconstructed tracks
+    for (const auto& track : myTracks) {
+      float drmin = 0.1;
+      // loop on the gen particles
+      for (auto g = genPartCollection->begin(); g != genPartCollection->end(); ++g) {
+        if (g->status() != 1)
+          continue;
+
+        if (std::abs(g->pdgId()) != 13)
+          continue;
+
+        if (g->charge() != track->charge())
+          continue;
+
+        float dR = reco::deltaR(*g, *track);
+
+        auto const& vtx = g->vertex();
+        auto const& myBeamSpot = beamSpot.position(vtx.z());
+        const auto& theptinv2 = 1 / g->pt() * g->pt();
+
+        if (dR < drmin) {
+          drmin = dR;
+
+          if (g->charge() > 0) {
+            genPosMuonPt_ = g->pt();
+            genPosMuonEta_ = g->eta();
+            genPosMuonPhi_ = g->phi();
+            //d0
+            genPosMuonD0_ = -(-(vtx.x() - myBeamSpot.x()) * g->py() + (vtx.y() - myBeamSpot.y()) * g->px()) / g->pt();
+            //dz
+            genPosMuonDz_ =
+                (vtx.z() - myBeamSpot.z()) -
+                ((vtx.x() - myBeamSpot.x()) * g->px() + (vtx.y() - myBeamSpot.y()) * g->py()) * g->pz() * theptinv2;
+          } else {
+            genNegMuonPt_ = g->pt();
+            genNegMuonEta_ = g->eta();
+            genNegMuonPhi_ = g->phi();
+            //d0
+            genNegMuonD0_ = (-(vtx.x() - myBeamSpot.x()) * g->py() + (vtx.y() - myBeamSpot.y()) * g->px()) / g->pt();
+            //dz
+            genNegMuonDz_ =
+                (vtx.z() - myBeamSpot.z()) -
+                ((vtx.x() - myBeamSpot.x()) * g->px() + (vtx.y() - myBeamSpot.y()) * g->py()) * g->pz() * theptinv2;
+          }
+        } else {
+          continue;
+        }
+      }
+    }
+  }
+
+  // if all cuts are passed fill the TTree
+  tree_->Fill();
+}
+
+//_________________________________________________________________________________
+double SagittaBiasNtuplizer::openingAngle(const reco::Track* trk1, const reco::Track* trk2) {
+  math::XYZVector vec1(trk1->px(), trk1->py(), trk1->pz());
+  math::XYZVector vec2(trk2->px(), trk2->py(), trk2->pz());
+  return vec1.Dot(vec2) / (vec1.R() * vec2.R());
+}
+
+/*
+double SagittaBiasNtuplizer::openingAngle(const reco::Track& track1, const reco::Track& track2) {
+  double dPhi = std::abs(track1.phi() - track2.phi());
+  double dEta = std::abs(track1.eta() - track2.eta());
+  double deltaR = reco::deltaR(track1, track2);
+  double openingAngle = 2 * std::atan2(std::sqrt(std::sin(dEta / 2) * std::sin(dEta / 2) + std::sinh(dPhi / 2) * std::sinh(dPhi / 2)), std::sqrt(1 - deltaR * deltaR));
```

Comment:
It is commented out, but...
```suggestion
  double deltaR2 = reco::deltaR2(track1, track2);
  double openingAngle = 2 * std::atan2(std::sqrt(std::sin(dEta / 2) * std::sin(dEta / 2) + std::sinh(dPhi / 2) * std::sinh(dPhi / 2)), std::sqrt(1 - deltaR2));
```